File: app/src/main/java/com/mxt/anitrend/base/custom/view/text/SpoilerTagTextView.java
package com.mxt.anitrend.base.custom.view.text;

import android.content.Context;
import android.util.AttributeSet;

import androidx.databinding.BindingAdapter;

import com.mxt.anitrend.R;
import com.mxt.anitrend.util.CompatUtil;

public class SpoilerTagTextView extends SingleLineTextView {

    public SpoilerTagTextView(Context context) {
        super(context);
    }

    public SpoilerTagTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SpoilerTagTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @BindingAdapter({"isSpoiler"})
    public static void setIsSpoiler(SingleLineTextView view, Boolean isSpoiler) {
        if (isSpoiler) view.setTextColor(CompatUtil.INSTANCE.getColor(view.getContext(), R.color.colorStateOrange));
    }
}

File: app/src/main/java/com/mxt/anitrend/base/custom/view/text/SpoilerTagTextView.kt
package com.mxt.anitrend.base.custom.view.text

import android.content.Context
import android.util.AttributeSet
import com.mxt.anitrend.R
import com.mxt.anitrend.util.CompatUtil

class SpoilerTagTextView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : SingleLineTextView(context, attrs, defStyleAttr) {

    companion object {
        @JvmStatic
        fun setIsSpoiler(view: SingleLineTextView, isSpoiler: Boolean?) {
            if (isSpoiler == true)
                view.setTextColor(CompatUtil.getColor(view.context, R.color.colorStateOrange))
        }
    }
}
